[17.0] [MIG] mrp_production_quant_manual_assign

MeritxellAForgeFlow opened the pull request:
Standard migration to 17.0

Depends on https://github.com/OCA/stock-logistics-warehouse/pull/2071

@ForgeFlow

yostashiro committed:
[ADD] mrp_production_quant_manual_assign

oca-travis committed:
[UPD] Update mrp_production_quant_manual_assign.pot

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

LoisRForgeFlow committed:
[14.0][MIG] mrp_production_quant_manual_assign

LoisRForgeFlow committed:
[IMP] mrp_production_quant_manual_assign: select lots to be consumed

this feature is only useful for MOs of final product that are
tracked by serial numbers.

oca-travis committed:
[UPD] Update mrp_production_quant_manual_assign.pot

OCA-git-bot committed:
[UPD] README.rst

ThiagoMForgeFlow committed:
[MIG] mrp_production_quant_manual_assign: Migration to 15.0

alexis-via committed:
[MIG] mrp_production_quant_manual_assign to v16

Fix bug on multi-uom when writing on stock.move.line

Commit:
[UPD] Update mrp_production_quant_manual_assign.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/es/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/it/

Commit:
Added translation using Weblate (French)

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/fr/

liuhehe1995 committed:
Added translation using Weblate (Japanese)

liuhehe1995 committed:
Translated using Weblate (Japanese)

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/ja/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/pt_BR/

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (7 of 7 strings)

Translation: manufacture-16.0/manufacture-16.0-mrp_production_quant_manual_assign
Translate-URL: https://translation.odoo-community.org/projects/manufacture-16-0/manufacture-16-0-mrp_production_quant_manual_assign/hr/

MeritxellAForgeFlow committed:
[IMP] mrp_production_quant_manual_assign: pre-commit auto fixes

LoisRForgeFlow commented on mrp_production_quant_manual_assign/wizards/assign_manual_quants.py:
```diff
@@ -53,18 +53,18 @@ def assign_quants(self):
             for ml in move.move_line_ids:
                 if ml.lot_id in lots_to_consume:
-                    ml.qty_done = ml.reserved_uom_qty
+                    ml.quantity = ml.reserved_uom_qty
```
reserved_uom_qty doesn't exist anymore in move lines in v17

MeritxellAForgeFlow replied:
fixed!

LoisRForgeFlow commented:
/ocabot migration mrp_production_quant_manual_assign

LoisRForgeFlow commented on mrp_production_quant_manual_assign/wizards/assign_manual_quants.py:
```diff
@@ -0,0 +1,70 @@
+            ).mapped("lot_id")
+            for ml in move.move_line_ids:
+                if ml.lot_id in lots_to_consume:
+                    ml.quantity = ml.product_uom_qty
```
AttributeError: 'stock.move.line' object has no attribute 'product_uom_qty'

LoisRForgeFlow commented on mrp_production_quant_manual_assign/wizards/assign_manual_quants.py:
```diff
@@ -0,0 +1,70 @@
+                if ml.lot_id in lots_to_consume:
+                    ml.quantity = ml.product_uom_qty
+                elif float_is_zero(
+                    ml.product_uom_qty, precision_digits=precision_digits
```
AttributeError: 'stock.move.line' object has no attribute 'product_uom_qty'

LoisRForgeFlow reviewed (changes requested):
There are still usages of non existing v17 columns. Could you check what I found and make sure that there isn't others?

MeritxellAForgeFlow committed:
[17.0][MIG] mrp_production_quant_manual_assign: Migration to 17.0

DavidJForgeFlow commented on mrp_production_quant_manual_assign/models/stock_move.py:
```diff
@@ -0,0 +1,20 @@
+    def _should_bypass_set_qty_producing(self):
+        res = super()._should_bypass_set_qty_producing()
+        if self.has_tracking != "none" and float_is_zero(
+            self.quantity_done, precision_rounding=self.product_uom.rounding
```
`            self.quantity, precision_rounding=self.product_uom.rounding`

MeritxellAForgeFlow replied:
sorry, i missed that one!

DavidJForgeFlow committed:
[IMP] mrp_production_quant_manual_assign

Adapt the wizard to work with the new picked field added in v17.0. Also added test cases to match the standard behaviors

LoisRForgeFlow reviewed (approved):
Functional test :+1:

LoisRForgeFlow commented:
/ocabot merge nobump
